connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/decoder/ActionDecoder.java

```diff
@@ -0,0 +1,122 @@
+package com.redhat.cloud.notifications.connector.email.decoder;
+
+import com.redhat.cloud.notifications.ingress.ActionOut;
+import com.redhat.cloud.notifications.ingress.Application;
+import com.redhat.cloud.notifications.ingress.Bundle;
+import com.redhat.cloud.notifications.ingress.Context;
+import com.redhat.cloud.notifications.ingress.Event;
+import com.redhat.cloud.notifications.ingress.EventType;
+import com.redhat.cloud.notifications.ingress.Metadata;
+import com.redhat.cloud.notifications.ingress.Payload;
+import com.redhat.cloud.notifications.ingress.Source;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+public class ActionDecoder {
```

Comment:
Is this class different from `Parser`?
https://github.com/RedHatInsights/notifications-backend/blob/3b34486d027298d253dd146f3d99021529182609/engine/src/main/java/com/redhat/cloud/notifications/utils/ActionParser.java#L13

Reply:
It is, because the engine sends details about the bundle, application and event type in a `source` key of the payload, which the parser rejects because its "an unknown key". I asked @josejulio and told me that I might as well create some ad hoc solution, since the parser might end up going away...

Reply:
I did, but didn't imagined it would get too complicated :sweat_smile: .
Could we use our base parser and fill in the gaps / extend to only parse the `source` part?

Reply:
I'll try to extend the parser, yes. Maybe I can override it :+1:

Reply:
IMO Implemeting `ActionOut` in the library itself would be better at this point instead of duplicating all the base parsing

Reply:
We finally ended up removing the `action` key from the payload that is sent to the connector. There is no need to implement this anymore... even though I thought about it when I was in the middle of it :sweat: